storage/storage-interface/src/lib.rs

```diff
@@ -534,6 +530,67 @@ pub trait DbWriter: Send + Sync {
         latest_in_memory_state: StateDelta,
         state_updates_until_last_checkpoint: Option<ShardedStateUpdates>,
         sharded_state_cache: Option<&ShardedStateCache>,
+    ) -> Result<()> {
+        // For reconfig suffix.
+        if ledger_info_with_sigs.is_none() && txns_to_commit.is_empty() {
+            return Ok(());
+        }
+
+        if !txns_to_commit.is_empty() {
```

Comment:
for this API, the txns should never be empty?

Reply:
There's a unit test in the executor crate that tests that case. Maybe we can remove after your PR.